### PR TITLE
e2e framework

### PR DESCRIPTION
**Pull request by eranra**

### CODE REVIEW

File: pkg/test/e2e/utils.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	clientSet, err := corev1client.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientSet, err
+}
+
+func LogsFromPods(pods corev1.PodList, clientSet *corev1client.CoreV1Client, namespace string) string {
+	logs := ""
```
Comment (ronensc) [inline pkg/test/e2e/utils.go:229]: Consider using `strings.Builder`. It should be more efficient.
https://www.calhoun.io/concatenating-and-building-strings-in-go/

File: pkg/test/e2e/utils.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		req := clientSet.Pods(namespace).GetLogs(pod.GetName(), &v1.PodLogOptions{Follow: false})
+		podLogs, err := req.Stream(context.TODO())
+		if err != nil {
+			// skipping
+			continue
+		}
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, podLogs)
+		if err != nil {
+			// skipping
```
Comment (ronensc) [inline pkg/test/e2e/utils.go:240]: Don't we want to log the error?

File: pkg/test/e2e/pipline/pipline_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	pipelineFeature := features.New("FLP/pipeline").WithLabel("env", "dev").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// Add any additional setup code and resources here
+			//client, err := cfg.NewClient()
+			//if err != nil {
+			//	t.Fatal(err)
+			//}
+			//if err := client.Resources().Create(ctx, resource); err != nil {
+			//	t.Fatal(err)
+			//}
```
[the remark this answers is not in the document]
Comment (eranra) [reply pkg/test/e2e/pipline/pipline_test.go:48]: removed the commented code

File: pkg/test/e2e/utils.go [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/envfuncs"
+)
+
+type namespaceContextKey string
+
+func Main(m *testing.M, TestEnv *env.Environment) {
```
[the remark this answers is not in the document]
Comment (eranra) [reply pkg/test/e2e/utils.go:50]: updated to `testEnv`

File: pkg/test/e2e/utils.go [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+	(*TestEnv).Finish(
+		envfuncs.DeleteNamespace(namespace),
+		envfuncs.DestroyKindCluster(kindClusterName),
+	)
+	os.Exit((*TestEnv).Run(m))
+}
+
+func e2eCreateKindCluster(clusterName string) env.Func {
+	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		fmt.Printf("====> Creating KIND cluster - %s\n", clusterName)
```
[the remark this answers is not in the document]
Comment (eranra) [reply pkg/test/e2e/utils.go:74]: @ronensc this is intended to make sure we always sprint those "high level" steps out and not using the log

File: pkg/test/e2e/utils.go [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+		fmt.Printf("====> Done.\n")
+		return newCtx, err
+	}
+}
+
+func e2eBuildAndLoadImageIntoKind(dockerImg, dockerTag, clusterName string) env.Func {
+	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		e := gexe.New()
+		fmt.Printf("====> building docker image - %s:%s\n", dockerImg, dockerTag)
+		p := e.RunProc(fmt.Sprintf(`/bin/sh -c "cd $(git rev-parse --show-toplevel); OCI_RUNTIME=docker DOCKER_IMG=%s DOCKER_TAG=%s make build-image"`,
```
[the remark this answers is not in the document]
Comment (eranra) [reply pkg/test/e2e/utils.go:85]: added a comment

File: pkg/test/e2e/utils.go [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+		if !acceptedK8sTypes.MatchString(groupVersionKind.Kind) {
+		} else {
+			resources = append(resources, resource)
+		}
+	}
+	return resources
+}
+
+func GetConfigSet(confFileName string) (*corev1client.CoreV1Client, error) {
```
[the remark this answers is not in the document]
Comment (eranra) [reply pkg/test/e2e/utils.go:214]: name update --- I am using **coreV1Client**

File: pkg/test/e2e/utils.go (hunk 1 above)
Comment (eranra) [reply pkg/test/e2e/utils.go:229]: updated :-)

File: pkg/test/e2e/utils.go (hunk 2 above)
Comment (eranra) [reply pkg/test/e2e/utils.go:240]: added log.Errorf

**Comment by eranra** [comment]
@KalmanMeth your assumption that kind exist in version XXX and deployed doesn't stand for the CI/CD process we fail here export KIND=$GOBIN/kind-v0.11.1 in the script ... this need to be changed

**Review by KalmanMeth — APPROVED** [review]

**Pull request merged**